File: 2-class-and-methods/1-from-struct.java
public class Main {
  // * 주의: 프로그램 진입점은
  // 처음으로 선언된 클래스의 static void main(String[] args) 함수이다.
  public static void main(String[] args) {
    Account acc = new Account();
    acc.name = "John";
    acc.password = "1234";

    System.out.println("Name: " + acc.name);
    System.out.println("Password: " + acc.password);

    acc.changePassword("5678");
    System.out.println("New password: " + acc.password);
  }
}

public class Account {
  String name;
  String password;

  // 클래스는 구조체와 함수를 합친 것이라고 볼 수 있다.
  // 그래서 클래스 아래에 함수를 선언할 수 있다.
  // 이 함수를 메서드라고 부른다.
  void changePassword(String newPassword) {
    password = newPassword;
  }
}

File: 2-java-defining-class/1-from-struct.c
#include <stdio.h>
#include <string.h>

// 계정 정보를 담는 구조체 생성
struct account {
  char name[50];
  char password[50];
} typedef account_t;

void change_password(struct account *acc, char *new_password) {
  strcpy(acc->password, new_password);
}

void file_change_password();

int main() {
  // 구조체 값 메모리에 생성
  account_t acc;
  // 구조체 값 초기화
  strcpy(acc.name, "admin");
  strcpy(acc.password, "admin");

  printf("Name: %s\n", acc.name);
  printf("Password: %s\n", acc.password);

  change_password(&acc, "new_password");
  printf("New password: %s\n", acc.password);
  return 0;
}
